### PR TITLE
Reduce requested bbox from all of the bbox to 10% of it.

### PR DESCRIPTION
**Pull request by RemiDesgrange**
When checking a layer. All the layer was requested with 10x10px image. 

We reduced the requested bbox to 10% of the bbox announced by the layer.

### CODE REVIEW

File: owscheck.py [hunk 1]
```diff
@@ -195,3 +196,8 @@ def get_service(self):
 
     def get_layer_names(self):
         return self._layer_names
+
+    def _reduced_bbox(self, bbox):
+        return [b*0.1 for f in bbox]
```
Comment (fvanderbiest) [inline owscheck.py:201]: Sure it works ?
eg there is no data outside 48.8, 1.34, 51.08, 4.28...

**Review by fvanderbiest — CHANGES_REQUESTED** [review]

File: owscheck.py (hunk 1 above)
Comment (fvanderbiest) [reply owscheck.py]:
```
>>> bbox = [ 48.8, 1.34, 51.08, 4.28 ]
>>> [b*0.1 for b in bbox]
[4.88, 0.134, 5.108, 0.428]
```
Resulting bbox is out of layer definition

Comment (fvanderbiest) [reply owscheck.py]: If we are to take 10% of the `[left, bottom, right, top]` bbox, then the computation should rather be 
`[left + 0.95*(right-left), bottom + 0.95*(top-bottom), right - 0.95*(right-left), top - 0.95*(top-bottom) ]`

Comment (fvanderbiest) [reply owscheck.py]: And really, I'm wondering if 10% is still not too big. Why not just 1% ? 
As long as we have a `200 OK` at the end ..

File: owscheck.py [hunk 2]
```diff
@@ -195,3 +196,13 @@ def get_service(self):
 
     def get_layer_names(self):
         return self._layer_names
+
+    def _reduced_bbox(self, bbox):
+        xmin, ymin, xmax, ymax = bbox
+        p = [xmin+0.49*(xmax-xmin),
+             ymin+0.49*(ymax-ymin),
+             xmax-0.49*(xmax-xmin),
+             ymax-0.49*(ymax-ymin)]
```
Comment (RemiDesgrange) [inline owscheck.py:205]: 0.49 does not mean reducing but 49% If we go over 50% then "xmin" become higher than "xmax".

To demonstrate "visually" copy this code in the JS pan of this [code pen](https://codepen.io/mina-tokyo/pen/dypKemy) (remove every JS).

```js
const zoom = 7;
const source = ''

const map = L.map('example-map');
L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
        maxZoom: 19,
        attribution: '&copy; <a href="https://openstreetmap.org/copyright">OpenStreetMap contributors</a>'
      }).addTo(map);
map.setView([45,5], zoom);


var originalBound = [[2.018463747207982, 44.597830110962].reverse(), [4.546589166293389, 46.815108752174545].reverse()];

var tenpercentBound = [[3.2572452025598313, 45.684296645156145].reverse(), [3.3078077109415394, 45.7286422179804].reverse()]

// create an orange rectangle
L.rectangle(originalBound, {color: "#ff7800", weight: 1}).addTo(map);
L.rectangle(tenpercentBound, {color: "#fffff", weight: 10}).addTo(map);

// zoom the map to the rectangle bounds
map.fitBounds(originalBound);
```

Comment (fvanderbiest) [reply owscheck.py]: > 0.49 does not mean reducing but 49% If we go over 50% then "xmin" become higher than "xmax".

Sure. Stupid me ;-)

**Comment by fvanderbiest** [comment]
Then we should do the same for the WFS bbox ... (10 lines lower)

File: owscheck.py (hunk 2 above)
Comment (RemiDesgrange) [reply owscheck.py]: I had to take a sheet of paper and draw it to be sure ;-) not practicing (basic) math makes me very rusty  !

**Comment by fvanderbiest** [comment]
Thanks !

**Pull request merged**